storage/src/device.rs

```diff
@@ -448,15 +456,21 @@ impl BlobChunkInfo for BlobIoChunk {
     }
 
     fn id(&self) -> u32 {
-        self.as_base().id()
+        match self {
+            Self::Address(_, index) => *index,
+            _ => self.as_base().id(),
+        }
     }
 
     fn blob_index(&self) -> u32 {
         self.as_base().blob_index()
     }
 
     fn compress_offset(&self) -> u64 {
-        self.as_base().compress_offset()
+        match self {
+            Self::Address(_, _) => 0,
```

Comment:
Which code could path trigger the panic?

Reply:
Aha, got it. Debug may trigger the panic.

Reply:
Could you please help to add a comments for it? Otherwise the code smells badly:)

Reply:
sure